server.js

```diff
@@ -102,11 +122,16 @@ function partialSearch(param, query) {
 
 // search
 app.get('/search', function(req, res) {
-  const searchResults = allTracks.filter(function(track) {
-    return partialSearch(track.artist, req.query.str) || partialSearch(track.album, req.query.str) || partialSearch(track.title, req.query.str);
+  searchService.search({"query" : {"*": [req.query.str]}}, function(err, results) {
+    if(err) {
+      console.log("Error executing search", q, err);
+    }
+    var hits = [];
```

Comment:
probably change this to a `let` unless we absolutely need hoisting in this scenario (doesn't look like it)